src/trading/orders.se

```diff
@@ -272,10 +272,10 @@ def fillOrder(orderId: address, orderType: int256, market: address, orderOutcome
     require(tx.gasprice <= self.orders[market][orderOutcome][orderType][orderId].gasPrice)
     fill = 0
     if orderType == BID:
-        # FIXME: we can't use safeSub here because we might be subtracting a negative, but we need to protect against overflows
+        # We can't use safeSub here because it disallows subtracting negative numbers. Worst case here is an operation of 2**254 - 1 which won't overflow
         fill = sharesFilled + safeFxpDiv(tokensFilled, self.orders[market][orderOutcome][orderType][orderId].fxpPrice - market.getMinDisplayPrice())
     if orderType == ASK:
-        # FIXME: we can't use safeSub here because we might be subtracting a negative, but we need to protect against overflows
+        # We can't use safeSub here because it disallows subtracting negative numbers. Worst case here is an operation of 2**254 - 1 which won't overflow
```

Comment:
Consider mentioning where the constraints are coming from in the comment.  I believe fxpPrice is constrained at storage write time (`saveOrder`) to `market.minValue` and `market.maxValue` and mention that `market.minValue + market.maxValue` is constrained at market creation time to `2**254`.

Normally I'm fairly against comments in code, preferring to try and improve code readability via verbose variable and function names but in this particular case I think there is value to auditors to make the assumptions very clear as well as where those assumptions are sourced from.

Alternatively, we could just add some `require(...)` statements at the top of this function that validate the above assertions, which would probably be the more correct thing to do.

```
require(self.orders[market][orderOutcome][orderType][orderId].fxpPrice <= market.getMaxDisplayPrice())
require(self.orders[market][orderOutcome][orderType][orderId].fxpPrice >= market.getMinDisplayPrice())
require(market.getMaxDisplayPrice() + market.getMinDisplayPrice < 2**254)
```

The purpose of these requires would not be because we believe there is an attack vector, but to ensure that future changes to the other parts of the code won't introduce a an attack vector into this code.